### ISSUE TITLE
torch.aten._scaled_dot_product_flash_attention_for_cpu "Rank Mismatch for Output. Expected: 4 Got: 5"

### ISSUE THREAD

**Issue by Alex-Vasile** [iree-org/iree#21070]
### What happened?

This version with static dims does not compile:
```mlir
module @module {
    func.func @static(%arg0: !torch.vtensor<[3,4,64,192],f32>, %arg1: !torch.vtensor<[3,4,64,192],f32>, %arg2: !torch.vtensor<[3,4,64,192],f32>, %arg3: !torch.tensor<[8,49152],f32>, %arg4: !torch.vtensor<[3,1,64,64],f32>) -> !torch.vtensor<[3,4,64,192],f32> attributes {torch.assume_strict_symbolic_shapes} {
        %float0. = torch.constant.float 0.000000e+00
        %false = torch.constant.bool false
        %none = torch.constant.none
        %237:2 = torch.operator "torch.aten._scaled_dot_product_flash_attention_for_cpu"(%arg0, %arg1, %arg2, %float0., %false, %arg4, %none) : (!torch.vtensor<[3,4,64,192],f32>, !torch.vtensor<[3,4,64,192],f32>, !torch.vtensor<[3,4,64,192],f32>, !torch.float, !torch.bool, !torch.vtensor<[3,1,64,64],f32>, !torch.none) -> (!torch.vtensor<[3,4,64,192],f32>, !torch.vtensor<[3,4,64],f32>) 
        return %237#0 : !torch.vtensor<[3,4,64,192],f32>
    }
}
```
It produces the following error message:
```
repro_static.mlir:6:18: error: Rank Mismatch for Output. Expected: 4 Got: 5
        %237:2 = torch.operator "torch.aten._scaled_dot_product_flash_attention_for_cpu"(%arg0, %arg1, %arg2, %float0., %false, %arg4, %none) : (!torch.vtensor<[3,4,64,192],f32>, !torch.vtensor<[3,4,64,192],f32>, !torch.vtensor<[3,4,64,192],f32>, !torch.float, !torch.bool, !torch.vtensor<[3,1,64,64],f32>, !torch.none) -> (!torch.vtensor<[3,4,64,192],f32>, !torch.vtensor<[3,4,64],f32>) 
                 ^
repro_static.mlir:6:18: note: see current operation: 
%32:3 = "iree_linalg_ext.online_attention"(%16, %17, %18, %3, %19, %29, %30, %31) <{decomposition_config = {pv_attrs = {lowering_config = #iree_gpu.lowering_config<{subgroup_basis = [[1, 1, 1, 1, 1, 1], [0, 1, 2, 3, 5]], thread = [0, 0, 0, 4, 0], thread_basis = [[1, 1, 1, 1, 1, 64], [2, 1, 0, 4, 5]]}>}, qk_attrs = {lowering_config = #iree_gpu.lowering_config<{subgroup_basis = [[1, 1, 1, 1, 1, 1], [0, 1, 2, 4, 5]], thread = [0, 0, 0, 4, 0], thread_basis = [[1, 1, 1, 1, 1, 64], [2, 1, 0, 4, 5]]}>}}, indexing_maps = [affine_map<(d0, d1, d2, d3, d4, d5) -> (d0, d1, d2, d4)>, affine_map<(d0, d1, d2, d3, d4, d5) -> (d0, d1, d5, d4)>, affine_map<(d0, d1, d2, d3, d4, d5) -> (d0, d1, d5, d3)>, affine_map<(d0, d1, d2, d3, d4, d5) -> ()>, affine_map<(d0, d1, d2, d3, d4, d5) -> (d0, d2, d5)>, affine_map<(d0, d1, d2, d3, d4, d5) -> (d0, d1, d2, d3)>, affine_map<(d0, d1, d2, d3, d4, d5) -> (d0, d1, d2)>, affine_map<(d0, d1, d2, d3, d4, d5) -> (d0, d1, d2)>]}> ({
^bb0(%arg19: f32):
  "iree_linalg_ext.yield"(%arg19) : (f32) -> ()
}) {lowering_config = #iree_gpu.lowering_config<{partial_reduction = [0, 0, 0, 0, 0, 0], workgroup = [1, 1, 1, 0, 0, 0]}>} : (tensor<1x1x1x192xf32>, tensor<1x1x64x192xf32>, tensor<1x1x64x192xf32>, f32, tensor<1x1x64xf32>, tensor<1x1x1x192x64xf32>, tensor<1x1x1x64xf32>, tensor<1x1x1x64xf32>) -> (tensor<1x1x1x192xf32>, tensor<1x1x1xf32>, tensor<1x1x1xf32>)
repro_static.mlir:6:18: error: failed to run translation of source executable to target executable for backend #hal.executable.target<"rocm", "rocm-hsaco-fb", {abi = "hip", iree.gpu.target = #iree_gpu.target<arch = "gfx942", features = "", wgp = <compute =  fp64|fp32|fp16|int64|int32|int16|int8, storage =  b64|b32|b16|b8, subgroup =  shuffle|arithmetic, dot =  dp4xi8toi32, mma = [<MFMA_F32_16x16x16_BF16>, <MFMA_F32_32x32x8_BF16>, <MFMA_F32_16x16x32_F8E5M2FNUZ>, <MFMA_F32_16x16x32_F8E5M2FNUZ_F8E4M3FNUZ>, <MFMA_F32_16x16x32_F8E4M3FNUZ>, <MFMA_F32_16x16x32_F8E4M3FNUZ_F8E5M2FNUZ>, <MFMA_F32_32x32x16_F8E5M2FNUZ>, <MFMA_F32_32x32x16_F8E5M2FNUZ_F8E4M3FNUZ>, <MFMA_F32_32x32x16_F8E4M3FNUZ>, <MFMA_F32_32x32x16_F8E4M3FNUZ_F8E5M2FNUZ>, <MFMA_I32_16x16x32_I8>, <MFMA_I32_32x32x16_I8>, <MFMA_F64_16x16x4_F64>, <MFMA_F32_16x16x4_F32>, <MFMA_F32_16x16x16_F16>, <MFMA_F32_32x32x8_F16>], subgroup_size_choices = [64], max_workgroup_sizes = [1024, 1024, 1024], max_thread_count_per_workgroup = 1024, max_workgroup_memory_bytes = 65536, max_workgroup_counts = [2147483647, 2147483647, 2147483647], max_load_instruction_bits = 128, simds_per_wgp = 4, vgpr_space_bits = 16384>>, iree_codegen.default_tuning_spec = #rocm.builtin.tuning_module<"iree_default_tuning_spec_gfx942.mlir">, ukernels = "none"}>
        %237:2 = torch.operator "torch.aten._scaled_dot_product_flash_attention_for_cpu"(%arg0, %arg1, %arg2, %float0., %false, %arg4, %none) : (!torch.vtensor<[3,4,64,192],f32>, !torch.vtensor<[3,4,64,192],f32>, !torch.vtensor<[3,4,64,192],f32>, !torch.float, !torch.bool, !torch.vtensor<[3,1,64,64],f32>, !torch.none) -> (!torch.vtensor<[3,4,64,192],f32>, !torch.vtensor<[3,4,64],f32>) 
```

Same code, but with a dynamic dim compiles:
```mlir
module @module {
    func.func @dynamic(%arg0: !torch.vtensor<[3,4,?,192],f32>, %arg1: !torch.vtensor<[3,4,?,192],f32>, %arg2: !torch.vtensor<[3,4,?,192],f32>, %arg3: !torch.tensor<[8,49152],f32>, %arg4: !torch.vtensor<[3,1,?,?],f32>) -> !torch.vtensor<[3,4,?,192],f32> attributes {torch.assume_strict_symbolic_shapes} {
        %float0. = torch.constant.float 0.000000e+00
        %false = torch.constant.bool false
        %none = torch.constant.none
        %237:2 = torch.operator "torch.aten._scaled_dot_product_flash_attention_for_cpu"(%arg0, %arg1, %arg2, %float0., %false, %arg4, %none) : (!torch.vtensor<[3,4,?,192],f32>, !torch.vtensor<[3,4,?,192],f32>, !torch.vtensor<[3,4,?,192],f32>, !torch.float, !torch.bool, !torch.vtensor<[3,1,?,?],f32>, !torch.none) -> (!torch.vtensor<[3,4,?,192],f32>, !torch.vtensor<[3,4,?],f32>) 
        return %237#0 : !torch.vtensor<[3,4,?,192],f32>
    }
}
```

### Steps to reproduce your issue

```
iree-compile case.mlir \
  -o=case.vmfb --iree-hal-target-device=hip \
  --iree-hip-target=gfx942 --iree-opt-level=O3 \
  --iree-hal-indirect-command-buffers=true \
  --iree-stream-resource-memory-model=discrete \
  --iree-hal-memoization=true
```

### What component(s) does this issue relate to?

Compiler

### Version information

`iree-3.5.0rc20250610`
